test/mitmproxy/test_command.py

```diff
@@ -356,20 +391,19 @@ def test_parse_partial(self):
             tctx.master.addons.add(TAddon())
             for s, expected, expectedremain in tests:
                 current, remain = tctx.master.commands.parse_partial(s)
-                assert current == expected
-                assert expectedremain == remain
+                assert (s, current, expectedremain) == (s, expected, remain)
 
 
 def test_simple():
     with taddons.context() as tctx:
         c = command.CommandManager(tctx.master)
         a = TAddon()
         c.add("one.two", a.cmd1)
-        assert(c.commands["one.two"].help == "cmd1 help")
-        assert(c.execute("one.two foo") == "ret foo")
-        assert(c.execute("one.two \"foo\"") == "ret foo")
-        assert(c.execute("one.two \"foo bar\"") == "ret \"foo bar\"")
-        assert(c.call("one.two", "foo") == "ret foo")
+        assert (c.commands["one.two"].help == "cmd1 help")
+        assert (c.execute("one.two foo") == "ret foo")
+        assert (c.execute("one.two \"foo\"") == "ret foo")
+        assert (c.execute("one.two \"foo bar\"") == "ret foo bar")
```

Comment:
So... this might not be correct. If we passed the string between quotes, we want it to come back between quotes when there is a space in the string. This signals that the params were treated as a single thing (in this case as a single string). 
This change re-introduces the bug I was trying to fix in the first place.

Try the following command:

`: set view_fiter=~bq "(?i)[^\"]ser"`

The application will say the filter is invalid, when in fact it is supposed to be valid.